proto/storage/cluster.proto

```diff
@@ -17,6 +17,29 @@ enum ClusterType {
     OPENSHIFT4_CLUSTER       = 5;
 }
 
+// ClusterMetadata contains metadata information about the cluster infrastructure.
+message ClusterMetadata {
+    enum Type {
+        UNSPECIFIED            = 0;
+        AKS                    = 1;
+        ARO                    = 2;
+        EKS                    = 3;
+        GKE                    = 4;
+        OPENSHIFT_SELF_MANAGED = 5;
+        ROSA                   = 6;
+    }
+
+    Type type   = 1 [(gogoproto.moretags) = 'search:"Cluster Type"'];
```

Comment:
I assume there is a distinct reason to make this searchable?

Reply:
See the mock above that I linked. We are planning to make this a new column in the cluster list.

Reply:
@connorgorman if you prefer I can leave out the search tag for now as well, I don't think it's strictly required.

Reply:
Its okay to add it